### ISSUE TITLE
I can't use my dataset train the model.

### ISSUE THREAD

**Issue by psnow** [mrharicot/monodepth#101]
Hi mrharicot,
    First of all thank you for your work.
    I am using a 3D movie as input , but the resulting disparity map is all black. My video frame is about 10000, were placed in two folders.
    my cmd is :python monodepth_main.py --model train --model_name my_demo_model --data_path /home/ddd/deeplearn/data/film/tframes/train/ --filenames_file /home/ddd/deeplearn/data/film/tframes/train/train.txt --log_director ./log/

![0](https://user-images.githubusercontent.com/6346340/35261258-59377fc0-004a-11e8-95b6-76eb0bd64e14.jpg)
![0](https://user-images.githubusercontent.com/6346340/35261264-5e15720e-004a-11e8-8626-4e4c8ed82c57.jpg)
![1](https://user-images.githubusercontent.com/6346340/35261271-6738b526-004a-11e8-9464-5379622d8b36.jpg)
![1](https://user-images.githubusercontent.com/6346340/35261277-6a55a0ca-004a-11e8-8558-6ae719bbaae8.jpg)

my filenames_file is like this:
left/3410.jpg right/3410.jpg
left/845.jpg right/845.jpg
left/4133.jpg right/4133.jpg
left/2800.jpg right/2800.jpg
left/1.jpg right/1.jpg
left/173.jpg right/173.jpg
left/3063.jpg right/3063.jpg
left/3783.jpg right/3783.jpg
left/2013.jpg right/2013.jpg
left/3183.jpg right/3183.jpg
left/195.jpg right/195.jpg
......

the total_loss:
![image](https://user-images.githubusercontent.com/6346340/35261368-c982ac8c-004a-11e8-8d03-ba9bf221a277.png)
![image](https://user-images.githubusercontent.com/6346340/35261390-deba98c6-004a-11e8-815e-bb1afe0b5f2f.png)

Looking forward th your help.

**Comment by mrharicot** [comment]
Hi,
3D movies are different as they do not have a fixed baseline, and also use negative disparities (converging cameras) to strengthen the effect.
This can be seen in your second example, where things move both left _and_ right.
You could try to change the [sigmoid in the disparity layer](https://github.com/mrharicot/monodepth/blob/master/monodepth_model.py#L124) to a tanh to allow for negative disparities:
`disp = 0.5 * self.conv(x, 2, 3, 1, tf.nn.tanh)`
The scenes vary a lot though, so it might be difficult for the network to learn from scratch.
You can have a look at this paper: https://arxiv.org/abs/1604.03650
Where they trained on 3d movies

**Comment by meijie0401** [comment]
Hi, thanks for your amazing code. I have a similar problem when training on my own 'fish dataset'. I have 5400 image pairs(256x256). The following two are examples. When I train the network from scratch, the total loss also didn't converge but some losses converged. What should I do to make it converge? Thanks!!

![0002595_20180630T155000 008-0800_GO-2400C-PGE+09-88-35 jpg_rectified-1 00](https://user-images.githubusercontent.com/37011125/67343484-447bcf00-f4ea-11e9-80d3-695439e4628b.jpg)
![0002595_20180630T155000 010-0800_GO-2400C-PGE+09-88-CA jpg_rectified-1 00 jpg-ht](https://user-images.githubusercontent.com/37011125/67343485-447bcf00-f4ea-11e9-9371-df5d03cabab3.jpg)
![Screenshot from 2019-10-22 16-37-56](https://user-images.githubusercontent.com/37011125/67343560-7c831200-f4ea-11e9-984a-5f964c61b849.png)
![Screenshot from 2019-10-22 16-38-17](https://user-images.githubusercontent.com/37011125/67343561-7c831200-f4ea-11e9-92fe-4f0c9f25bc7c.png)
![Screenshot from 2019-10-22 16-38-25](https://user-images.githubusercontent.com/37011125/67343563-7c831200-f4ea-11e9-8701-43c02c2f82f3.png)
![Screenshot from 2019-10-22 16-38-32](https://user-images.githubusercontent.com/37011125/67343564-7d1ba880-f4ea-11e9-89fb-654305d707d9.png)
![Screenshot from 2019-10-22 16-38-54](https://user-images.githubusercontent.com/37011125/67343565-7d1ba880-f4ea-11e9-81dc-4a238a091bcd.png)
![Screenshot from 2019-10-22 16-39-01](https://user-images.githubusercontent.com/37011125/67343566-7d1ba880-f4ea-11e9-80f2-b7a3048ca344.png)

**Comment by mrharicot** [comment]
Hi,
Are all your training pairs similar to these ones?
The main problem is that your stereo image has both positive and negative disparities.
Are all your pairs similar to this one?
